src/Client.ts

```diff
@@ -125,15 +125,23 @@ export default class Client
 
         if (request.headers instanceof Headers) {
```

Comment:
Looking at this in retrospect, we can probably simplify this to always work with a headers object. On line 122 of the original file, instead of checking for the existence of `request.headers`, we can just do this:

```typescript
request.headers = new Headers(request.headers);
```

This will work no matter if `request.headers` is undefined or one of the allowed values, basically casting it into a headers object. This will allow us to get rid of the if statements below and always just work with the headers object.

Reply:
@DASPRiD take a look at my changes.  If request.headers is undefined and you call new Headers(request.headers); it will throw an error in IE and Edge so I put a if else in.

Reply:
IE and Edge are not a problem, since this is a server-side only library, only NodeJS is important :)

Reply:
it does not have to be server side only.  how about
`        request.headers = new Headers(request.headers === undefined ? {} : request.headers);`

Reply:
We are using nodejs-side only libraries though.